decidim/decidim
# Fix linking to invariable image URLs

#### :tophat: What? Why?
Not all environments support creating "variable URLs" for all image types. Let's say the admin user would allow uploading image files that cannot be converted using ImageMagick.

In this case, the file conversion fails because of an exception of type `ActiveStorage::InvariableError` is thrown.

How the user would see this is that they cannot e.g. preview their proposal or see the proposal page because it is trying to create the thumbnail URLs for these images. This PR fixes the issue by falling back to the original URL of that file.

#### :pushpin: Related Issues
- Fixes #10237

#### Testing
- Within the development_app's `config/environments/development.rb`, add the following configuration: `config.active_storage.variable_content_types = %w(image/png)`
- Make sure `image/jpeg` is allowed from the Decidim system panel (it should be if you haven't changed anything)
- Allow attachments for proposals
- Try to submit a proposal with a JPEG image or attachment
- See error (before this patch)
